### ISSUE TITLE
functionapp deployment failes with Getting scm site credentials for zip deployment

### ISSUE THREAD

**Issue by DanielHWe** [Azure/azure-cli#10207]
> ### `az feedback` auto-generates most of the information requested below, as of CLI version 2.0.70

**Describe the bug**
After updating from 2.0.57 to 2.0.70 my deploment of a zip to a functionapp does no more work. Error is: Getting scm site credentials for zip deployment.
Command is:
az functionapp deployment source config-zip  -g "XXX" -n "XXX" --src "deploy.zip"

**To Reproduce**
Use 
az functionapp deployment source config-zip  -g "XXX" -n "XXX" --src "deploy.zip"
on a deployed function app

**Expected behavior**

**Environment summary**
Windows 10
az -v shows
azure-cli                         2.0.70

command-modules-nspkg               2.0.3
core                              2.0.70
nspkg                              3.0.4
telemetry                          1.0.3

Python location 'C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\python.exe'
Extensions directory 'C:\Users\Wehrled\.azure\cliextensions'

Python (Windows) 3.6.6 (v3.6.6:4cf1f54eb7, Jun 27 2018, 02:47:15) [MSC v.1900 32 bit (Intel)]

Legal docs and information: aka.ms/AzureCliLegal

Your CLI is up-to-date.

**Additional context**
With a other pc running still 2.0.57 it still works fine.

**Comment by ahmedelnably** [comment]
@ankitkumarr can you please take a look?

@DanielHWe can you please provide the output of the command with the `--debug` flag?

**Comment by DanielHWe** [comment]
I switched from powershell to cmd getting more information:

Getting scm site credentials for zip deployment
Starting zip deployment. This operation can take a while to complete ...
The command failed with an unexpected error. Here is the traceback:
```
Expecting value: line 1 column 1 (char 0)
Traceback (most recent call last):
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-rtk110ve\knack\knack\cli.py", line 206, in invoke
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-rtk110ve\azure-cli-core\azure\cli\core\commands\__init__.py", line 603, in execute
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-rtk110ve\azure-cli-core\azure\cli\core\commands\__init__.py", line 661, in _run_jobs_serially
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-rtk110ve\azure-cli-core\azure\cli\core\commands\__init__.py", line 654, in _run_job
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-rtk110ve\six\six.py", line 693, in reraise
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-rtk110ve\azure-cli-core\azure\cli\core\commands\__init__.py", line 631, in _run_job
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-rtk110ve\azure-cli-core\azure\cli\core\commands\__init__.py", line 305, in __call__
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-rtk110ve\azure-cli-core\azure\cli\core\__init__.py", line 485, in default_command_handler
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-rtk110ve\azure-cli\azure\cli\command_modules\appservice\custom.py", line 296, in enable_zip_deploy_functionapp
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-rtk110ve\azure-cli\azure\cli\command_modules\appservice\custom.py", line 322, in enable_zip_deploy
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-rtk110ve\azure-cli\azure\cli\command_modules\appservice\custom.py", line 2326, in _check_zip_deployment_status
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-rtk110ve\requests\requests\models.py", line 897, in json
  File "C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\lib\json\__init__.py", line 354, in loads
    return _default_decoder.decode(s)
  File "C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\lib\json\decoder.py", line 339, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\lib\json\decoder.py", line 357, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

To open an issue, please run: 'az feedback'
```
with debug I got in the end:
```
Starting zip deployment. This operation can take a while to complete ...
urllib3.connectionpool : Starting new HTTPS connection (1): update20functionapp2019a3.scm.azurewebsites.net:443
urllib3.connectionpool : https://update20functionapp2019a3.scm.azurewebsites.net:443 "POST /api/zipdeploy?isAsync=true HTTP/1.1" 202 0
urllib3.connectionpool : Starting new HTTPS connection (1): update20functionapp2019a3.scm.azurewebsites.net:443
urllib3.connectionpool : https://update20functionapp2019a3.scm.azurewebsites.net:443 "GET /api/deployments/latest HTTP/1.1" 503 27
cli.azure.cli.core.util : The command failed with an unexpected error. Here is the traceback:

The command failed with an unexpected error. Here is the traceback:

cli.azure.cli.core.util : Expecting value: line 1 column 1 (char 0)
Traceback (most recent call last):
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-rtk110ve\knack\knack\cli.py", line 206, in invoke
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-rtk110ve\azure-cli-core\azure\cli\core\commands\__init__.py", line 603, in execute
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-rtk110ve\azure-cli-core\azure\cli\core\commands\__init__.py", line 661, in _run_jobs_serially
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-rtk110ve\azure-cli-core\azure\cli\core\commands\__init__.py", line 654, in _run_job
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-rtk110ve\six\six.py", line 693, in reraise
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-rtk110ve\azure-cli-core\azure\cli\core\commands\__init__.py", line 631, in _run_job
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-rtk110ve\azure-cli-core\azure\cli\core\commands\__init__.py", line 305, in __call__
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-rtk110ve\azure-cli-core\azure\cli\core\__init__.py", line 485, in default_command_handler
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-rtk110ve\azure-cli\azure\cli\command_modules\appservice\custom.py", line 296, in enable_zip_deploy_functionapp
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-rtk110ve\azure-cli\azure\cli\command_modules\appservice\custom.py", line 322, in enable_zip_deploy
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-rtk110ve\azure-cli\azure\cli\command_modules\appservice\custom.py", line 2326, in _check_zip_deployment_status
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-rtk110ve\requests\requests\models.py", line 897, in json
  File "C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\lib\json\__init__.py", line 354, in loads
    return _default_decoder.decode(s)
  File "C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\lib\json\decoder.py", line 339, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\lib\json\decoder.py", line 357, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
Expecting value: line 1 column 1 (char 0)
Traceback (most recent call last):
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-rtk110ve\knack\knack\cli.py", line 206, in invoke
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-rtk110ve\azure-cli-core\azure\cli\core\commands\__init__.py", line 603, in execute
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-rtk110ve\azure-cli-core\azure\cli\core\commands\__init__.py", line 661, in _run_jobs_serially
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-rtk110ve\azure-cli-core\azure\cli\core\commands\__init__.py", line 654, in _run_job
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-rtk110ve\six\six.py", line 693, in reraise
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-rtk110ve\azure-cli-core\azure\cli\core\commands\__init__.py", line 631, in _run_job
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-rtk110ve\azure-cli-core\azure\cli\core\commands\__init__.py", line 305, in __call__
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-rtk110ve\azure-cli-core\azure\cli\core\__init__.py", line 485, in default_command_handler
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-rtk110ve\azure-cli\azure\cli\command_modules\appservice\custom.py", line 296, in enable_zip_deploy_functionapp
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-rtk110ve\azure-cli\azure\cli\command_modules\appservice\custom.py", line 322, in enable_zip_deploy
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-rtk110ve\azure-cli\azure\cli\command_modules\appservice\custom.py", line 2326, in _check_zip_deployment_status
  File "C:\Users\VSSADM~1\AppData\Local\Temp\pip-install-rtk110ve\requests\requests\models.py", line 897, in json
  File "C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\lib\json\__init__.py", line 354, in loads
    return _default_decoder.decode(s)
  File "C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\lib\json\decoder.py", line 339, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\lib\json\decoder.py", line 357, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
cli.azure.cli.core.util :
To open an issue, please run: 'az feedback'
```

**Comment by ahmedelnably** [comment]
What is the SKU of the app you are trying to deploy to?

**Comment by DanielHWe** [comment]
Im not sure what SKU means in this context. I try to upload a function app (Runtime version: 1.0.12615.0 (~1)) by use of "az functionapp deployment source config-zip" to a resource in westeurope.

**Comment by ankitkumarr** [comment]
SKU is the type of plan you are on. Are you on a Consumption plan or an App Service Plan? Also what OS is your functionapp on, if you know? 

Or you can also share your function app name, and we can get the SKU and OS information. If you would prefer to share it anonymously, you can do so by following these steps https://github.com/Azure/azure-functions-host/wiki/Sharing-Your-Function-App-name-privately

**Comment by DanielHWe** [comment]
We are on a Consumption Plan. Id of function is 22d01435-a299-4ae7-884b-83b8a62b6f12.

I deploy from powershell on windows 10. The function is written in .Net Framework.

**Comment by DanielHWe** [comment]
Does still not work, any updates?

**Comment by ankitkumarr** [comment]
I apologize that this got lost somewhere. Would you mind sharing the ID along with an approximate time-period for me to dig this down.

I will prioritize this for this week and get back to you as soon as I have the above information.

**Comment by TeeNak** [comment]
I encountered same error when trying zip deploy after non zip deploy like `func azure functionapp publish xxx`.
In this case azure service returns error text which is not recognized as a json like following.
```
Deployment 'latest' not found.
```
Is this information helps?

**Comment by TeeNak** [comment]
I encountered same problem with azure cli 2.0.57.
But after updating azure cli tools to 2.0.75. it works fine without any problem.

**Comment by DanielHWe** [comment]
2.0.57 works fine, but 2.0.70 and 2.0.75 not (Some colleges have no problem with 2.0.70 on other systems). Also the problem is only in powershell. A cmd is working fine.

**Comment by saids** [comment]
I had the same issue and error, and on the PS it was outputting this info..

azure-cli                         2.0.77 *
command-modules-nspkg              2.0.3
core                              2.0.77 *
nspkg                              3.0.4
telemetry                          1.0.4

Python location 'C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\python.exe'
Extensions directory 'C:\Users\{USERNAME}\.azure\cliextensions'

Python (Windows) 3.6.6 (v3.6.6:4cf1f54eb7, Jun 27 2018, 02:47:15) [MSC v.1900 32 bit (Intel)]

Legal docs and information: aka.ms/AzureCliLegal

**You have 2 updates available. Consider updating your CLI installation. Instructions can be found at https://docs.microsoft.com/en-us/cli/azure/install-azure-cli**

did the upgrade on the cli 
**Invoke-WebRequest -Uri https://aka.ms/installazurecliwindows -OutFile .\AzureCLI.msi; Start-Process msiexec.exe -Wait -ArgumentList '/I AzureCLI.msi /quiet'**

after the installation did exit from powershell and reopened. Ran the script again to see if it generates the error. There were still same errors unfortunately..

**Comment by 4riel** [comment]
Same problem over here when using ps...
azure-cli  2.11.1
core 2.11.1
telemetry 1.0.5 *

**Comment by navba-MSFT** [comment]
@DanielHWe Apologies for the late reply. Could you please let me know if you are still facing the same issue with [most recent version](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli) of Az CLI? Awaiting you reply.

**Comment by saids** [comment]
Why MSFT answers now? It's too late!

**Comment by DanielHWe** [comment]
Hi, this issue is closed with the current version of azure cli, I can not reproduce it any more.

**Comment by saids** [comment]
@DanielHWe One day, It will come back!  It's a resilient bug :-)

**Comment by vipinanandcpp** [comment]
how did you just close this issue? it's just unbelievable Microsoft replies after 2 years and still can't replicate when half the world is facing this issue!!!!!! **az functionapp deployment source config-zip --src "build.zip" --name f-snowflakedbupdater-dv --resource-group datalink-dv-e2us-rg --verbose --build-remote true** This is the command we have been trying to run over the weekend. And let me make this crystal clear: We spent 90% time of the time deploying this thing and only 10% on testing.... it came to a point that the only way to get out of this error is to head out and grab a coffee which took me 15 minutes then that error morphed into **HTTPSConnectionPool(host='f-snowflakedbexecutor-dv.scm.azurewebsites.net', port=443): Max retries exceeded with url: /api/deployments/latest (Caused by SSLError(SSLError(1, '[SSL: TLSV1_ALERT_INTERNAL_ERROR] tlsv1 alert internal error (_ssl.c:2548)')))** and then I waited for 15 more minutes to actually see this finished successfully.
